File: backend/store/app/order/query/book_mapper.go
package query

import "store/app/domain/data"

func (Book) fromDataObject(b data.Book) Book {
	return Book{
		GoogleBookId:  b.GoogleBookId,
		Title:         b.Title,
		Subtitle:      b.Subtitle,
		Description:   b.Description,
		Authors:       b.Authors,
		AverageRating: b.AverageRating,
		RatingsCount:  b.RatingsCount,
		ThumbnailUrl:  b.ThumbnailUrl,
		PreviewUrl:    b.PreviewUrl,
		ReservedQty:   b.ReservedQty,
		OnhandQty:     b.OnhandQty,
	}
}

File: backend/ecommerce/src/main/java/com/bookstore/ecommerce/app/order/query/BookMapper.java
package com.bookstore.ecommerce.app.order.query;

public class BookMapper {
  public static Book fromDataObject(com.bookstore.ecommerce.app.domain.data.Book book) {
    return Book.builder().id(book.getId()).title(book.getTitle()).subTitle(book.getSubTitle())
        .description(book.getDescription()).build();
  }
}
